Abolish required fields in favor of more flexibility, set required fields in template?

Here's what I got:

fields:
  Carrier: [A-Z0-9]
  Attn: \W
  Phone: \s+(\d{3}-\d{3}-\d{4})
  Fax: \s+(\d{3}-\d{3}-\d{4})?
  MC Number: \s+\d+
  "BOL #": \s+[0-9A-Z]
  Commodity: \s+[A-Za-z]
  Reference: \s[.]
  Earliest Date: \s.
  Latest Date: \s.
  Buyer: \s.
  Email: [A-Za-z\.\@\s]
  Contact: \s.
  "PO #": \s+[0-9A-Z]
  "Pick Up #": \s+[0-9A-Z]
  "Delivery #": \s+[0-9A-Z]

keywords:
  - Company Name
  - MC Number

then I get this error:

AssertionError: Missing required key in template usa.got.logistics.english.yml /Users/raulgonzalez/Documents/yml_templates. Found ['Carrier', 'Attn', 'Phone', 'Fax', 'MC Number', 'BOL #', 'Commodity', 'Reference', 'Earliest Date', 'Latest Date', 'Buyer', 'Email', 'Contact', 'PO #', 'Pick Up #', 'Delivery #']

--- a/invoice2data/extract/invoice_template.py
+++ b/invoice2data/extract/invoice_template.py
@@ -170,11 +170,17 @@ class InvoiceTemplate(OrderedDict):
                 plugin_func.extract(self, optimized_str, output)
 
         # If required fields were found, return output, else log error.
-        if set(['date', 'amount', 'invoice_number', 'issuer']).issubset(output.keys()):
-            output['desc'] = 'Invoice %s from %s' % (
-                output['invoice_number'], self['issuer'])
-            logger.debug(output)
-            return output
+        if 'required-fields' not in self.keys():
+            required_fields = ['date', 'amount', 'invoice_number', 'issuer']
         else:
-            logger.error('Unable to match some fields:', output)
-            return None
+            required_fields = []
+            for v in self['required-fields']:
+                required_fields.append(v)
+        if set(required_fields).issubset(output.keys()):
+                output['desc'] = 'Invoice %s from %s' % (
+                    output['invoice_number'], self['issuer'])
+                logger.debug(output)
+                return output
+            else:
+                logger.error('Unable to match some fields:', output)
+                return None

--- a/invoice2data/extract/invoice_template.py
+++ b/invoice2data/extract/invoice_template.py
@@ -176,11 +176,12 @@ class InvoiceTemplate(OrderedDict):
             required_fields = []
             for v in self['required-fields']:
                 required_fields.append(v)
+        
         if set(required_fields).issubset(output.keys()):
                 output['desc'] = 'Invoice %s from %s' % (
                     output['invoice_number'], self['issuer'])
                 logger.debug(output)
                 return output
-            else:
-                logger.error('Unable to match some fields:', output)
-                return None
+        else:
+            logger.error('Unable to match some fields:', output)
+            return None

